Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -339,55 +339,168 @@ void NetworkStorageManager::writeOriginToFileIfNecessary(const WebCore::ClientOr
 #else
     UNUSED_PARAM(didWrite);
 #endif
+}
+
+void NetworkStorageManager::spaceGrantedForOrigin(const WebCore::ClientOrigin& origin, uint64_t spaceRequested)
+{
+    assertIsCurrent(workQueue());
+
+    updateLastModificationTimeForOrigin(origin);
+    if (!m_totalQuotaRatio)
+        return;
+
+    if (!m_totalQuota) {
+        std::optional<uint64_t> volumeCapacity;
+        if (m_volumeCapacityOverride)
+            volumeCapacity = m_volumeCapacityOverride;
+        else if (auto capacity = FileSystem::volumeCapacity(m_path))
+            volumeCapacity = WTF::roundUpToMultipleOf(defaultVolumeCapacityUnit, *capacity);
+        if (volumeCapacity)
+            m_totalQuota = *m_totalQuotaRatio * *volumeCapacity;
+        else
+            return;
+    }
 
-    if (m_quotaManager)
-        m_quotaManager->originVisited(origin);
+    if (m_totalUsage)
+        m_totalUsage = *m_totalUsage + spaceRequested;
+
+    if (!m_totalUsage || *m_totalUsage > *m_totalQuota)
```

Comment:
Why scheduling eviction if m_totalUsage is std::nullopt?

Reply:
To initialize `m_totalUsage` in `donePrepareForEviction`.